src/testdrive/src/action/skip_if.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use anyhow::Context as _;
+use async_trait::async_trait;
+
+use crate::action::{Action, Context, State};
+use crate::parser::BuiltinCommand;
+
+pub struct SkipIfAction {
+    query: String,
+    context: Context,
+}
+
+pub fn build_publish(
+    mut cmd: BuiltinCommand,
+    context: Context,
+) -> Result<SkipIfAction, anyhow::Error> {
+    let query = cmd.args.string("sql")?;
+    Ok(SkipIfAction { query, context })
+}
+
+#[async_trait]
+impl Action for SkipIfAction {
+    async fn undo(&self, _: &mut State) -> Result<(), anyhow::Error> {
+        Ok(())
+    }
+
+    async fn redo(&self, state: &mut State) -> Result<(), anyhow::Error> {
+        let stmt = state
+            .pgclient
+            .prepare(&self.query)
+            .await
+            .context("failed to prepare skip-if query")?;
+
+        let actual: Vec<_> = state
+            .pgclient
+            .query(&stmt, &[])
+            .await
+            .context("executing query failed")?
+            .into_iter()
+            .map(|row| crate::action::sql::decode_row(row, self.context.clone()))
+            .collect::<Result<_, _>>()?;
+
+        if vec![vec!["true".to_string()]] == actual {
```

Comment:
There's gotta be some way to get a real bool here no?

Reply:
I think if I avoid going through `decode_row`, but i want to re-use as much as possible

Reply:
I don't follow! I think `decode_row` is getting in the way. I think you just want to use rust-postgres's normal API for extracting a Rust `bool` from a SQL query that returned a bool.